src/network.cpp

```diff
@@ -266,141 +379,173 @@ namespace xiloader
-                {
-                    xiloader::console::output(xiloader::color::error, "Passwords did not match! Please try again.");
-                }
-            } while (g_Password != confirmed_password);
-
-            /* Clear the buffers */
-            memset(sendBuffer, 0, 33);
-            memset(recvBuffer, 0, 16);
-
-            /* Copy the new password into the buffer. */
-            memcpy(sendBuffer, g_Password.c_str(), 16);
+                xiloader::console::output(xiloader::color::success, "Successfully logged in as %s!", globals::g_Username.c_str());
 
-            /* Send info to server and obtain response.. */
-            send(sock->s, sendBuffer, 16, 0);
-            recv(sock->s, recvBuffer, 16, 0);
+                sock->AccountId = ref<UINT32>(recvBuffer, 1);
+                std::memcpy(&globals::g_SessionHash, recvBuffer + 5, sizeof(globals::g_SessionHash));
 
-            /* Handle the final result. */
-            switch (recvBuffer[0])
+                shutdown(sock->s, SD_BOTH);
+                closesocket(sock->s);
+                sock->s = INVALID_SOCKET;
+                return true;
+            }
+            case 0x0002: // Error (Login)
+            {
+                xiloader::console::output(xiloader::color::error, "Failed to login. Invalid username or password.");
+                closesocket(sock->s);
+                sock->s = INVALID_SOCKET;
+                return false;
+            }
+            case 0x0003: // Success (Create Account)
             {
+                xiloader::console::output(xiloader::color::success, "Account successfully created!");
+                closesocket(sock->s);
+                sock->s = INVALID_SOCKET;
+                return false;
+            }
+            case 0x0004: // Error (Create Account)
+            {
+                xiloader::console::output(xiloader::color::error, "Failed to create the new account. Username already taken.");
+                closesocket(sock->s);
+                sock->s = INVALID_SOCKET;
+                return false;
+            }
             case 0x0006: // Success (Changed Password)
+            {
                 xiloader::console::output(xiloader::color::success, "Password updated successfully!");
                 std::cout << std::endl;
-                g_Password.clear();
+                globals::g_Password.clear();
                 closesocket(sock->s);
                 sock->s = INVALID_SOCKET;
                 return false;
-
+            }
             case 0x0007: // Error (Changed Password)
+            {
                 xiloader::console::output(xiloader::color::error, "Failed to change password.");
                 std::cout << std::endl;
-                g_Password.clear();
+                globals::g_Password.clear();
+                closesocket(sock->s);
+                sock->s = INVALID_SOCKET;
+                return false;
+            }
+            // Gap for previously used command codes
```

Comment:
Without much context I dont understand, can you flesh this comment out for idiots like me please?

Reply:
That comment was not new. I guess xiloader used to have more commands before I even entered the scene. I can remove it since it doesn't matter anymore.

Reply:
Changed the comment to reflect there are unused command codes